vendor: bump crlfmt a78e1c207bc0 to b3eff0b87c79

stevendanna opened the pull request:
This bumps the version of cockroachdb/crlfmt, pulling in the following
changes:

  - add -srcdir option similar to goimports
  - use printer options that match gofmt when simplifying
  - Fix link to CockroachDB Go coding guidelines
  - using go module to manage dependencies
  - add simplify from gofmt -s
  - Update README with usage info
  - set default tab width to 2

Additionally, this pulls in a new version of golang.org/x/tools:

   golang/tools@d8d6ddb...v0.1.0

A manual entry for golang.org/x/sys has been added to the bazel
WORKSPACE to avoid an older version from being pulled in during `make
bazel-generate`.

Release note: None

stevendanna committed:
vendor: bump crlfmt a78e1c207bc0 to b3eff0b87c79

This bumps the version of cockroachdb/crlfmt, pulling in the following
changes:

  - add -srcdir option similar to goimports
  - use printer options that match gofmt when simplifying
  - Fix link to CockroachDB Go coding guidelines
  - using go module to manage dependencies
  - add simplify from gofmt -s
  - Update README with usage info
  - set default tab width to 2

Additionally, this pulls in a new version of golang.org/x/tools:

   https://github.com/golang/tools/compare/d8d6ddbec6ee...v0.1.0

A manual entry for golang.org/x/sys has been added to the bazel
WORKSPACE to avoid an older version from being pulled in during `make
bazel-generate`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59575)
<!-- Reviewable:end -->

irfansharif commented:
Just checked it out locally (and rebasing off of `master` to pick up a recent bazel build breakage), this SHA ends up tripping up the bazel build (try `bazel build --color=yes @cockroach//pkg/cmd/cockroach-short --sandbox_debug`). I'm pretty sure this isn't the diff that's at fault, so looking now.

irfansharif commented:
Hm, I'm still not really sure why we're seeing this error now, but it's because of code changes in `x/sys` that we're now picking up. Specifically https://github.com/golang/sys/blob/22da62e12c0cd9c1da93581e1113ca4d82a5be14/execabs/execabs.go#L41-L43.

I think we'll have to update how we use `stringer` in the bazel sandbox, evidently some behaviour has changed that we were implicitly relying on. Essentially all instances of `org_golang_x_tools//cmd/stringer:stringer` in our codebase. I won't be able to investigate this week, but I know @rickystewart, @alan-mas, and others in #dev-inf are working around bazel as of late. So perhaps they'd be able to help.

stevendanna commented:
I can also take a look between onboarding sessions this week. Apologies that this didn't build out of the box, I had assumed that CI was running the bazel builds.

irfansharif commented:
Nothing to apologize for! The "bazel build" is currently optional in CI (and I think was recently suspended because of other issues). I'll defer to Ricky here, but if I'm reading correctly that this PR isn't immediately urgent, I'd personally prefer to try and address the /x/sys incompatibility first before merging (as opposed to breaking the bazel build, which a small minority of us try to use day-to-day).

stevendanna commented:
> if I'm reading correctly that this PR isn't immediately urgent

Yup, not urgent at all

stevendanna committed:
build,sql,util: use absolute path to go in stringer invocation

The go/tools package now raises an error if you try to execute go via
a relative path inside the current working directory. To get around
this, we resolve the absolute path to the go binary.

We use the cd+pwd method of finding the absolute path to avoid
problems with the available of GNU readlink or realpath on macOS.

Release note: None

stevendanna commented:
Added a commit with a "quick fix" for the bazel build problem.

rickystewart commented:
The "quick fix" seems fine to me.

stevendanna commented:
Rebased this on master and I think it should be good to go.

irfansharif commented on build/STRINGER.bzl:
```diff
@@ -2,16 +2,17 @@
       outs = [typ.lower() + "_string.go"],
       cmd = """
-         env PATH=`dirname $(location @go_sdk//:bin/go)` HOME=$(GENDIR) \
+         GO_REL_PATH=`dirname $(location @go_sdk//:bin/go)`
```
Mind adding a brief comment explaining why this was necessary?

stevendanna commented:
bors r=irfansharif
